pkg/ccl/sqlproxyccl/balancer/balancer.go

```diff
@@ -140,6 +167,22 @@ type Balancer struct {
 	// timeutil.DefaultTimeSource. Override with the TimeSource() option when
 	// calling NewBalancer.
 	timeSource timeutil.TimeSource
+
+	// rebalanceRate represents the rate of rebalancing connections.
+	rebalanceRate float32
+
+	// rebalanceDelay is the minimum amount of time that must elapse between
+	// attempts to rebalance a given tenant. Defaults to defaultRebalanceDelay.
+	rebalanceDelay time.Duration
```

Comment:
Thought: we could probably get rid of rebalanceDelay if we improve the behavior of rebalanceRate. Currently if we schedule two reconciles back to back, it will schedule connections*rebalanceRate transfers. If we limited the number of in progress transfers to rebalanceRate * connections, we could remove rebalanceDelay.

Alternatively we could change the definition to something like:
We can rebalance connections * rebalanceRate of a tenant's connections every second. If rebalances would exceed that rate, they are delayed until the next poll for idle connections.